tools/screenshot-test/src/app/firebase.service.ts

```diff
@@ -0,0 +1,198 @@
+import {Injectable} from '@angular/core';
+import * as firebase from 'firebase';
+
+const request = require('request');
+const config = require('../config.json');
+
+
+export class ScreenshotResult {
+  /** PR information: the pull request number */
+  prNumber: string;
+  /** PR information: the sha of the pull request commit */
+  sha: string;
+  /** PR information: The travis job ID */
+  travis: string;
+
+  /** Test result, the test names */
+  testnames: string[];
+  /** Test result: passed or failed. The value can be true if test failed but PR approved by user */
+  result: boolean;
+  /** Test results: passed or failed for each test */
+  results: Map<string, boolean> = new Map<string, boolean>();
+  /**
+   * Test approved: whether the test images are copied to goldens.
+   * The value is the commit SHA and date/time of approval.
+   */
+  approved: string;
+
+  githubStatus: any;
+
+  /** Viewing mode, can be flip, diff, side */
+  mode: 'diff' | 'side' | 'flip' = 'diff';
+  /** Viewing flipping, whether the image is test image or golden image */
+  flipping: boolean = false;
+  /** Viewing collapsed: whether the result should be collapsed/expanded */
+  collapse: boolean[];
```

Comment:
Each test result may be collapsed or expanded, so it's an array. 

`isFlipped` is the way to view image, and all the images are flipped at the same time, so we only need a single boolean.